src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
> From a protocol point of view, switching keyboard layout implies sending a new wl_keyboard::keymap event to all clients, and these clients are probably supposed to load a new keymap themselves.

Thats some very interesting info. Can we as a libinput user actually detect, which keyboard is sending a key and just switch the layout if the other is used?